src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs

```diff
@@ -38,8 +38,8 @@ public CommandLineApplication(bool throwOnUnexpectedArg = true)
         public List<string> RemainingArguments { get; private set; }
         public bool IsShowingInformation { get; protected set; }  // Is showing help or version?
         public Func<int> Invoke { get; set; }
-        public Func<string> VersionGetter { get; set; }
-
+        public Func<string> LongVersionGetter { get; set; }
+        public Func<string> ShortVersionGetter { get; set; }
```

Comment:
Yeah but it shouldn't. The version should be just line any other command/argument, not something special because it doesn't apply to subcommands